Mathlib/LinearAlgebra/Matrix/Charpoly/Coeff.lean

```diff
@@ -272,17 +276,17 @@ open LaurentPolynomial hiding C
 
 It has some advantages over the characteristic polynomial, including the fact that it can be
 extended to infinite dimensions (for appropriate operators). -/
-lemma Matrix.reverse_charpoly (M : Matrix n n R) :
-    M.charpoly.reverse = det (1 - (X : R[X]) • C.mapMatrix M) := by
+lemma reverse_charpoly (M : Matrix n n R) :
+    M.charpoly.reverse = det (1 - (X : R[X]) • M.map C) := by
```

Comment:
By now there are quite a number of results about this polynomial. I think it deserves a better name than `det_one_sub_X_smul_C`. The docstring of this lemma says that it is sometimes called "characteristic power series". So how about abstracting it into a definition called `charPowSeries`? Or possibly `charpolyRev`.

Reply:
Also, is `M.map (fun r => r • (X : R[X]))` better or worse than  ` `(X : R[X]) • M.map C`?

Reply:
@jcommelin Thanks, now done.

@eric-wieser I have no preference (beyond small effort to make a switch) but I'm happy to switch if you prefer.

Reply:
I have no preference either, but the fact that neither is the obvious choice isn't great for discoverability.